Build failure: pkgsStatic.lmdb

### Steps To Reproduce

Steps to reproduce the behavior:
1. Run `nix-build '<nixpkgs>' -A pkgsStatic.lmdb`

alternatively (the output is the same)

1. Run `nix-build -E '(with import <nixpkgs> { system = "x86_64-linux"; }; pkgsStatic.lmdb)'`

### Build log

```
unpacking sources
unpacking source archive /nix/store/4n9fq724aqaivhv8myrizzd75qqs7di1-source
source root is source
patching sources
applying patch /nix/store/vfsz428k08c21cnrh67yw2xc950z01wh-hardcoded-compiler.patch
patching file Makefile
applying patch /nix/store/33p9ajg7n3k71p7xkn2kxhghgis2rc8g-bin-ext.patch
patching file Makefile
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash prefix=\$\(out\) CC=x86_64-unknown-linux-musl-cc AR=x86_64-unknown-linux-musl-ar
x86_64-unknown-linux-musl-cc -pthread -O2 -g -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized   -c mdb.c
x86_64-unknown-linux-musl-cc -pthread -O2 -g -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized   -c midl.c
x86_64-unknown-linux-musl-ar rs liblmdb.a mdb.o midl.o
x86_64-unknown-linux-musl-ar: creating liblmdb.a
x86_64-unknown-linux-musl-cc -pthread -O2 -g -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized  -fPIC  -c mdb.c -o mdb.lo
x86_64-unknown-linux-musl-cc -pthread -O2 -g -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized  -fPIC  -c midl.c -o midl.lo
x86_64-unknown-linux-musl-cc  -pthread -shared -o liblmdb.so mdb.lo midl.lo 
/nix/store/i2p87r479cr697b4v3g2565ymghnqi8y-x86_64-unknown-linux-musl-binutils-2.40/bin/x86_64-unknown-linux-musl-ld: /nix/store/qx7k00a837fad50z9cgxjhp8vawm6s4k-x86_64-unknown-linux-musl-stage-final-gcc-12.2.0/lib/gcc/x86_64-unknown-linux-musl/12.2.0/crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
/nix/store/i2p87r479cr697b4v3g2565ymghnqi8y-x86_64-unknown-linux-musl-binutils-2.40/bin/x86_64-unknown-linux-musl-ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make: *** [Makefile:70: liblmdb.so] Error 1
```

### Additional context

Found this while trying to build `pkgsStatic.murmur` on my mac

I also tested this running on my nixos work computer instead of via a remote builder and got the same result.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@jb55 
@vcunat 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.5`
 - channels(<user>): `"darwin, home-manager-23.05.tar.gz"`
 - channels(root): `"nixpkgs-23.05-darwin"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Though the actual building is done by a remote builder with `x86_64-linux`.

--- a/pkgs/development/libraries/lmdb/default.nix
+++ b/pkgs/development/libraries/lmdb/default.nix
@@ -17,6 +17,11 @@ stdenv.mkDerivation rec {
   patches = [ ./hardcoded-compiler.patch ./bin-ext.patch ];
   patchFlags = [ "-p3" ];
 
+  # Don't attempt the .so if static, as it would fail.
+  postPatch = lib.optionalString stdenv.hostPlatform.isStatic ''
+    sed 's/^ILIBS\>.*/ILIBS = liblmdb.a/' -i Makefile
+  '';
+
   outputs = [ "bin" "out" "dev" ];
 
   buildInputs = lib.optional stdenv.hostPlatform.isWindows windows.pthreads;

